src/client.rs

```diff
@@ -91,6 +91,12 @@ pub fn connect_with_config<Req: IntoClientRequest>(
     config: Option<WebSocketConfig>,
 ) -> Result<(WebSocket<AutoStream>, Response)> {
     let request: Request = request.into_client_request()?;
+    // Copy all the fields from the initial reqeust **except** the URI. This will be used in the event of a redirection code
+    // Have to manually clone Method because there is one field that contains a Box,
+    // but in the case of normal request methods it is Copy
+    let request2 = Request::builder()
+        .method(request.method().clone())
+        .version(request.version());
```

Comment:
Alternatively we could make a requirement that `IntoClientRequest` also implements `Clone`, then we could just clone it and change the URI which probably would look a bit more elegant.

Reply:
Ah, got it. I just thought that what basically we want to have is to have the same request, but with a different URI, thus the semantics of `request.clone().set_uri(new_uri)`.

Reply:
Request isn't Clone so we'd still have to formulate it like that anyways